sdk/src/transaction.rs

```diff
@@ -246,6 +255,26 @@ impl Transaction {
             .collect())
     }
 
+    /// Replace all the signatures and pubkeys
+    pub fn replace_signatures(&mut self, signers: &[(Pubkey, Signature)]) -> Result<()> {
+        let num_required_signatures = self.message.header.num_required_signatures as usize;
+        if signers.len() != num_required_signatures
+            || self.signatures.len() != num_required_signatures
+            || self.message.account_keys.len() < num_required_signatures
```

Comment:
These checks seem outside of the scope of this method and could lead to confusing errors to the caller of `replace_signatures`. I think that it's fine to rely on our runtime to reject these errors. Thoughts?

Reply:
Having these checked up front rather then after the transaction is submitted would be very helpful.  I was also following the example of `get_signing_keypair_positions`.  Maybe we should generate more descriptive errors?

Reply:
Never mind, I see that by doing this you are avoiding a panic from a bad index. Looks fine 👍